### ISSUE TITLE
New Rule: `no-empty-images`

### ISSUE THREAD

**Issue by lumirlumir** [eslint/markdown#355]
### Rule details

Rule to disallow empty images

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
<!-- ERROR -->

![]()

![](#)

![image]()

<!-- OK -->

![](https://google.com/image.png)
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Hello,

I’d like to propose a new rule: `no-empty-images`.

In Markdown, it’s not always easy to spot when you’ve forgotten to provide a destination for an image. This is especially common when writing something in Markdown, where you might intend to insert an image and leave the destination to fill in later—only to forget to go back and complete it.

This often results in broken image links like the one below:

![Image](https://github.com/user-attachments/assets/a7894660-fb10-495b-bc78-a0a249165bc)

---

If this proposal is acceptable, I’d like to go ahead and submit a PR before May 10!

**Comment by nzakas** [comment]
Thanks for the suggestion. Do any existing markdown linters have a similar rule?

**Comment by lumirlumir** [comment]
@nzakas 

[`remark-lint`](https://github.com/remarkjs/remark-lint?tab=readme-ov-file#rules), part of the same ecosystem as [`mdast`](https://github.com/syntax-tree/mdast), includes a rule called [`remark-lint-no-empty-url`](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-empty-url).

This rule checks for empty `link` and `image` nodes, effectively combining the behavior of `@eslint/markdown`’s `no-empty-links` rule with a `no-empty-images` rule, which I proposed.

**Comment by nzakas** [comment]
Got it, thanks! This rule sounds like a good idea to me. I'd also say it should be recommended. :+1:

**Comment by lumirlumir** [comment]
I will complete the work by May 10👍
